### ISSUE TITLE
Supervised Loss Computation 

### ISSUE THREAD

**Issue by gmbss0** [vanderschaarlab/mlforhealthlabpub#1]
I have a question regarding the implementation of the **supervised loss training**. This is computed in the code as follows:

 # 2. Supervised loss    
G_loss_S = tf.losses.mean_squared_error(H[:,1:,:], **H_hat_supervise**[:,1:,:]), where H_hat_supervise is computed as:
H_hat_supervise = supervisor(H, T)

Should not the supervised loss be computed on the latent representation coming from the generator (at least this is how I understood the paper):

E_hat = generator(Z, T)
**H_hat = supervisor(E_hat, T)** ?
